File: bio-informatics/challenges/reverse-complement-string.js
const reverseComplementString = (dnaString) => {
    let reversedDnaString = "";

    for (let i = 0; i < dnaString.length; i++) {
        const nucleobase = dnaString[i];
        switch (nucleobase) {
            case "A":
                reversedDnaString += "T";
                break;
            case "T":
                reversedDnaString += "A";
                break;
            case "G":
                reversedDnaString += "C";
                break;
            case "C":
                reversedDnaString += "G";
                break;
        }
    }

    return(reversedDnaString);
}

File: bio-informatics/challenges/python/utils/reverse_complement_string.py
def reverse_complement_string(dnaString):
  reversed_complement_string = ""

  paired_nucleotides = {
    "A": "T",
    "T": "A",
    "C": "G",
    "G": "C"
  }

  for nucleotide in dnaString:
    try:
      reversed_complement_string += paired_nucleotides.get(nucleotide)
    except:
      raise Exception(f"invalid nucleotide in string: {nucleotide}")

  return reversed_complement_string
  